### ISSUE TITLE
feat(cli): improve error messages

### ISSUE THREAD

**Issue by jamiehaywood** [JamieMason/syncpack#132]
## Description

I have a syncpack config:

```
module.exports = {
  customTypes: [],
  dependencyTypes: ['dev', 'peer', 'prod', 'workspace'],
  semverGroups: [
    {
      range: '^',
      dependencies: ['@my-internal-pkg/**'],
      packages: ['**'],
      isIgnored: true,
      dependencyTypes: ['dependencies', 'resolutions', 'overrides', 'pnpmOverrides', 'workspace'],
    },
  ],
  semverRange: '',
  sortAz: [
    'contributors',
    'dependencies',
    'devDependencies',
    'keywords',
    'peerDependencies',
    'resolutions',
    'scripts',
  ],
  sortFirst: ['name', 'description', 'version', 'author'],
  versionGroups: [],
}
```

When I run `yarn syncpack fix-mismatches`

I get an error from Zod validation:

```
cause: ZodError: [
    {
      "code": "unrecognized_keys",
      "keys": [
        "range"
      ],
      "path": [
        "semverGroups",
        0
      ],
      "message": "Unrecognized key(s) in object: 'range'"
    }
  ]
```

I'm not sure why this would be the case, because the docs and the code allow for the `semverGroups` object to have a key of `range`?

## Suggested Solution

Not sure if this is a `cosmiconfig` issue or something else.

**Comment by JamieMason** [comment]
Hey Jamie, it's because the group needs to be either one (ignored) or the other (set to a range) so with them both being set it's not clear what to do. Zod will be removed in the next release and I'll do some work to improve the error messages a little too.

**Comment by JamieMason** [comment]
In [syncpack@10.0.0](https://github.com/JamieMason/syncpack/releases/tag/10.0.0) @jamiehaywood's intellisense work will now catch this scenario, but some runtime validation with useful error messages would be good. Leaving this issue open for that side of things.

**Comment by JamieMason** [comment]
Released in [10.6.1](https://github.com/JamieMason/syncpack/releases/tag/10.6.1). Any problems, please let me know. 

If you find syncpack useful, please star the project or help us spread the word to other Developers.
